src/main/java/org/opensearch/geospatial/ip2geo/jobscheduler/DatasourceRunner.java

```diff
@@ -118,161 +105,42 @@ public void runJob(final ScheduledJobParameter jobParameter, final JobExecutionC
+                if (lock == null) {
+                    return;
                 }
-
                 try {
-                    if (client.admin()
-                        .indices()
-                        .prepareDelete(index)
-                        .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN)
-                        .execute()
-                        .actionGet(timeout)
-                        .isAcknowledged()) {
-                        deletedIndices.add(index);
-                    } else {
-                        log.error("Failed to delete an index [{}]", index);
-                    }
-                } catch (Exception e) {
-                    log.error("Failed to delete an index [{}]", index, e);
+                    updateDatasource(jobParameter);
+                } finally {
+                    lockService.release(
+                        lock,
+                        ActionListener.wrap(released -> {}, exception -> { log.error("Failed to release lock [{}]", lock, exception); })
+                    );
                 }
-            }
-            if (!deletedIndices.isEmpty()) {
-                parameter.getIndices().removeAll(deletedIndices);
-                DatasourceHelper.updateDatasource(client, parameter, timeout);
-            }
-        } catch (Exception e) {
-            log.error("Failed to delete old indices for {}", parameter.getId(), e);
-        }
+            }, exception -> { log.error("Failed to acquire lock for job [{}]", jobParameter.getName(), exception); }));
+        };
     }
 
-    /**
-     * Update GeoIP data internal
-     *
-     * @param jobParameter
-     * @throws Exception
-     */
-    private void updateDatasource(final Datasource jobParameter) throws Exception {
-        if (!DatasourceState.AVAILABLE.equals(jobParameter.getState())) {
-            log.error("Invalid datasource state. Expecting {} but received {}", DatasourceState.AVAILABLE, jobParameter.getState());
-            jobParameter.disable();
-            jobParameter.getUpdateStats().setLastFailedAt(Instant.now());
-            DatasourceHelper.updateDatasource(client, jobParameter, timeout);
-            return;
-        }
-
-        URL url = new URL(jobParameter.getEndpoint());
-        DatasourceManifest manifest = DatasourceManifest.Builder.build(url);
-
-        if (skipUpdate(jobParameter, manifest)) {
-            log.info("Skipping GeoIP database update. Update is not required for {}", jobParameter.getId());
-            jobParameter.getUpdateStats().setLastSkippedAt(Instant.now());
-            DatasourceHelper.updateDatasource(client, jobParameter, timeout);
+    @VisibleForTesting
+    protected void updateDatasource(final ScheduledJobParameter jobParameter) throws IOException {
+        Datasource datasource = datasourceHelper.getDatasource(jobParameter.getName());
+        if (datasource == null) {
+            log.info("Datasource[{}] is already deleted", jobParameter.getName());
```

Comment:
nit: shall we use term doesn't exists instead of deleted? The first doesn't assume it was previously created

Reply:
It should have been created. Otherwise it won't get updated. Anyway I think it is better to say `doesn't exist`. Will update.